src/main/java/org/sasanlabs/configuration/VulnerableAppConfiguration.java

```diff
@@ -123,4 +135,29 @@ public JdbcTemplate applicationJdbcTemplate(
             @Qualifier("applicationDataSource") DataSource applicationDataSource) {
         return new JdbcTemplate(applicationDataSource);
     }
+
+    /**
+     * Customized MultipartFilter bean disables default max upload size for multipart files and
+     * their overall requests, for select paths. See {@link
+     * UnrestrictedFileUpload#getVulnerablePayloadLevel10()} for usage.
+     */
+    @Bean
+    @Order(0)
+    public MultipartFilter multipartFilter() {
+        class CustomMF extends MultipartFilter {
+            @Override
+            protected MultipartResolver lookupMultipartResolver(HttpServletRequest request) {
+                if (MAX_FILE_UPLOAD_SIZE_OVERRIDE_PATHS.contains(request.getServletPath())) {
```

Comment:
This way even simple url with CONTROLLER_PATH will match this filter. Does checking other way around be better?

Reply:
I'm not sure I understand this comment. The servlet path must exactly match
`"/" + UnrestrictedFileUpload.CONTROLLER_PATH + "/" + LevelConstants.LEVEL_10`
for condition to evaluate to true.

Reply:
MAX_FILE_UPLOAD_SIZE_OVERRIDE_PATHS is a list of string values. I set it up this  way so later we can add to it if necessary. The check here is `contains()`, as it checks if there is a string in the list, which is equal to the path. In this context, this is really an "equals" check, with any of the values of the list. Which is, for now, just one possible value.
I can change this to be just a single value, but I believe, the behavior is already what you are suggesting.

Reply:
aauch ... My mistake.